src/common/qat.c

```diff
@@ -348,37 +393,54 @@ qat_dc_compress_async(
 			numBuffers,
 			&bufferMetaSize);
 
-	/* Allocate source buffer */
-	if (status == CPA_STATUS_SUCCESS)
-		status = mem_alloc_contig(
-				(void *)&pBufferMetaSrc, bufferMetaSize, 1);
+	if (!SVM_enabled) {
+		/** Allocate source buffer */
+		if (status == CPA_STATUS_SUCCESS)
+			status = mem_alloc_contig(
+					(void *)&pBufferMetaSrc, bufferMetaSize, 1);
 
+		if (status == CPA_STATUS_SUCCESS)
+			status = mem_alloc_contig((void *)&pSrcBuffer, srcLen, 1);
+
+		/** Allocate destination buffer the same size as source buffer */
+		if (status == CPA_STATUS_SUCCESS)
+			status = mem_alloc_contig(
+					(void *)&pBufferMetaDst, bufferMetaSize, 1);
+
+		if (status == CPA_STATUS_SUCCESS)
+			status = mem_alloc_contig((void *)&pDstBuffer, dstLen, 1);
+
+		/** Copy source into buffer */
+		memcpy(pSrcBuffer, src, srcLen);
+	} else {
+		if (status == CPA_STATUS_SUCCESS) {
```

Comment:
Add a comment here - If the SVM (Shared Virtual Memory) is enabled, the virtual memory address can be sent to QAT device directly, allocating physical contiguous memory is not required.

Reply:
done